lightly/models/modules/heads.py

```diff
@@ -370,14 +370,20 @@ class SwaVPrototypes(ProjectionHead):
 
     """
     def __init__(self,
-                 input_dim: int = 128,
-                 n_prototypes: int = 3000):
-        super(SwaVPrototypes, self).__init__([])
-        self.layers = nn.Linear(input_dim, n_prototypes, bias=False)
-
-    def normalize(self):
-        """Normalizes the prototypes so that they are on the unit sphere."""
-        utils.normalize_weight(self.layers.weight)
+                input_dim: int = 128,
+                n_prototypes: Union[List[int], int] = 3000):
+        super(SwaVPrototypes, self).__init__()
+        self.number_of_heads = len(n_prototypes) if isinstance(n_prototypes, list) else 1
+        #Default to a list of 1 if n_prototypes is an int.
+        self.n_prototypes = n_prototypes if isinstance(n_prototypes, list) else [n_prototypes]
+        for layerNum, k in enumerate(self.n_prototypes):
+            self.add_module("prototypes" + str(layerNum), nn.Linear(input_dim, k, bias = False))
+
+    def forward(self, x):
+        out = []
+        for head in range(self.number_of_heads):
+            out.append(getattr(self, "prototypes" + str(head))(x))
+        return out
```

Comment:
This changes the behaviour of  the `SwaVPrototypes` head and might break existing code. Before this change it returned a single tensor and now it returns a list of tensors.
 
I would propose to keep the old behaviour if users specify `n_prototypes` with a single number. If `n_prototypes` is a list, the forward function should also return a list. This could be implemented by adding a `self.single_head = isinstance(n_prototypes, int)` line in the `__init__` method and checking the property in the forward function.

Reply:
Hey @guarin. So just to clarify, the output should be a tensor if n_prototypes is an int else, return a list. I think this should be easy to implement. Am I correct ?

Reply:
Thanks for implementing the feature :)